agent/php_memcached.c

```diff
@@ -0,0 +1,18 @@
+#include "php_memcached.h"
+#include "nr_datastore_instance.h"
+#include "php_agent.h"
+
+nr_datastore_instance_t* nr_php_memcached_create_datastore_instance(
+    const char* host,
+    zend_long port) {
+  nr_datastore_instance_t* instance = NULL;
+  if (port == 0) { // local socket
+    instance = nr_datastore_instance_create("localhost", host, NULL);
```

Comment:
1. Why is `host` used for `port_path_or_id` argument when `port` is unknown?
2. Does `0 == port` if and only if `host` == `"localhost"`? Is there a case when `0 == port` but `host` != `localhost`?

Reply:
The parameters are explained here: https://www.php.net/manual/en/memcached.addserver.php#refsect1-memcached.addserver-parameters

As far as I can tell, 0 port is always and only used for accessing a localhost socket filepath

Reply:
Knowing that, consider changing the name of the argument to `host_or_socket_name`. This name would match the doc string as well as be less confusing when used in this context.

Reply:
done in [d0aabeb](https://github.com/newrelic/newrelic-php-agent/pull/958/commits/d0aabeb7846be00f2b0cb74ebecf5953aa45a84a)